src/modules/menu/components/activeLayers/MenuActiveLayersList.vue

```diff
@@ -26,13 +26,22 @@ const aLayerIsDragged = ref(false)
 const showLayerDescriptionForLayer = ref(null)
 const showLayerDetailIndex = ref(null)
 const layerDetailFocusMoveButton = ref(null)
-
+const showLayerDescriptionIndex = ref(null)
 const store = useStore()
 // Users are used to have layers ordered top to bottom (the first layer is on top), but we store them in the opposite order.
 // So here we swap the order of this array to match the desired order on the UI
 const activeLayers = computed(() => store.state.layers.activeLayers.slice().reverse())
 const reverseIndex = (index) => store.state.layers.activeLayers.length - 1 - index
 
+watch(activeLayers, (newActiveLayer) => {
+    if (showLayerDescriptionIndex.value !== null) {
+        onShowLayerDescription(
+            newActiveLayer[showLayerDescriptionIndex.value],
+            showLayerDescriptionIndex.value
+        )
+    }
+})
+
```

Comment:
We should be watching a language change, not a layer change, here. This will be called if I add or remove a layer from the active layers, which is not what we want.
For example, if I add the public transports stops layer, click somewhere, and add the RBD building layer, the popup will now show the building layer which is not what we want.

We might need to do a quick check with `getIndexOfActiveLayerById` to find the layer again after the lang change, do something like 
```
if (showLayerDescription.value) {
showLayerDescription.value = activeLayers.value[getIndexOfActiveLayerById(showLayerDescription.value)]
}
```

Reply:
We discussed about this with Pascal, and we have found a more elegant solution, but it is quite difficult to explain. He will either send you a gist (or make a commit) on this issue.

Also, I made a mistake about the active layers side effects in this case :)